packages/cli/commons/src/errors/preconditionError.ts

```diff
@@ -9,7 +9,7 @@ export enum PreconditionErrorCategory {
 }
 
 export interface PreconditionErrorOptions {
-  category?: PreconditionErrorCategory;
+  category?: PreconditionErrorCategory | string;
```

Comment:
Why string?

Reply:
To allow a plugin to define its own precondition error